submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepository.kt

```diff
@@ -141,9 +144,16 @@ class SubmissionDocDataRepository(
                 filter.type?.let { add(where("$SUB_SECTION.$SEC_TYPE").`is`(it)) }
                 filter.rTimeFrom?.let { add(where(SUB_RELEASE_TIME).gte(it.toInstant())) }
                 filter.rTimeTo?.let { add(where(SUB_RELEASE_TIME).lte(it.toInstant())) }
-                filter.keywords?.let { add(where(SUB_TITLE).regex("(?i).*$it.*")) }
+                filter.keywords?.let { add(keywordsCriteria(it)) }
                 filter.released?.let { add(where(SUB_RELEASED).`is`(it)) }
             }.build().toTypedArray()
+
+        private fun keywordsCriteria(keywords: String) = Criteria().orOperator(
```

Comment:
I was pretty sure we had an integration test for this and I was wondering how was it passing in mongo so I took a look:

https://github.com/EBIBioStudies/biostudies-backend-services/blob/master/submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/query/SubmissionListApiTest.kt#L153

We have the right case but we're making the wrong test, searching by accNo instead of keyword 🤦‍♂️. Could you please take a look?